site/js/ChannelsActor.js

```diff
@@ -118,14 +125,14 @@ export default class ChannelsActor extends Actor {
     return this.channels.find(c => c.name === channelName) || null
   }
 
-  viewChannel(channelID) {
+  async viewChannel(channelID) {
     this.activeChannelID = channelID
-    this.socket.emit('view channel', channelID)
+    //this.socket.emit('view channel', channelID) // TODO use api for this
```

Comment:
>use api for this

Keep in mind that channel ID is stored on the server per-socket, not per-user. It actually looks like you've already written the code which handles `view channel`, so I'm wondering why this is commented out.

Reply:
Yeah, I uncommented this later afaik...?

Reply:
Oh crap maybe not. It should be uncommented as I reimplemented it!